spec/statement.md

```diff
@@ -0,0 +1,176 @@
+
+## Syntax
+
+### ScssStatements
+
+<x><pre>
+**ScssStatements**      ::= (Statement ';'?¹)* Statement?
+</pre></x>
+
+1. This production is mandatory unless the previous `Statement` is a
+`LoudComment`, `SilentComment`, or ends in a `Block`.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement [IndentSame])* Statement
+</pre></x>
+
+[IndentSame]: #indentation
+
+The `Statement` productions may not include newlines outside of `IndentSame`
+productions.
+
+### Stylesheet
+
+<x><pre>
+**Stylesheet**          ::= U+FEFF? ([ScssStatements] | [IndentedStatements])¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+1. Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements]
+**Block**          ::= (ScssBlock | IndentedBlock)¹
+</pre></x>
+
+[IndentMore]: #indentation
+
+1. Only the production for the current syntax is valid.
+
+### Comments
+
+#### LoudComment
+
+<x><pre>
+**ScssLoudComment**          ::= '/\*' (.\*¹ | Interpolation)\* '\*/'
+**IndentedLoudCommentNest**  ::= [IndentMore] ((.\*² | Interpolation)\*
+&#32;                            [IndentSame])\* (.\*² | Interpolation)\*
+**IndentedLoudComment**      ::= '/\*' (.\*² | Interpolation)\*
+&#32;                            IndentedLoudCommentNest?
+**LoudComment**              ::= (ScssLoudComment | IndentedLoudComment)³
+</pre></x>
+
+1. This may not contain `#{` or `*/`.
+2. This may not contain `#{` or newlines.
+3. Only the production for the current syntax is valid.
+
+#### SilentComment
+
+<x><pre>
+**ScssSilentComment**          ::= '//' .\*¹
+**IndentedSilentComment**      ::= '//' .\*²
+**SilentComment**              ::= (ScssSilentComment | IndentedSilentComment)³
+</pre></x>
+
+1. This may not contain newlines.
+2. This may not contain newlines outside of [IndentSame] productions.
+3. Only the production for the current syntax is valid.
+
+#### WhitespaceComment
+
+<x><pre>
+**ScssWhitespaceComment**          ::= '//' .\*¹ | '/\*' .\*² '\*/'
+**IndentedWhitespaceComment**      ::= '/\*' .\*² '\*/'
+**WhitespaceComment**³             ::= ScssWhitespaceComment
+&#32;                                | IndentedWhitespaceComment
+</pre></x>
+
+1. This may not contain newlines.
+2. This may not contain `*/`.
+3. Only the production for the current syntax is valid.
+
+### Whitespace
+
+<x><pre>
+**LineBreak**               ::= CarriageReturn | LineFeed | FormFeed
+**ScssWhitespace**          ::= LineBreak | Space | Tab
+**IndentedWhitespace**      ::= Space | Tab
+**Whitespace**              ::= (ScssWhitespace | IndentedWhitespace)¹
+&#32;                         | [WhitespaceComment]
+</pre></x>
+
+1. Only the production for the current syntax is valid.
```

Comment:
Potentially a simpler way to define this would be:

```suggestion
**LineBreak**  ::= CarriageReturn | LineFeed | FormFeed
**Whitespace** ::= LineBreak¹ | Space | Tab | [WhitespaceComment]
</pre></x>

1: This is not allowed in the indented syntax.
```